### ISSUE TITLE
Type variable renamer does not work for mutually recursive functions

### ISSUE THREAD

**Issue by jstolarek** [links-lang/links#882]
**EDIT**: see comments below for an updated explanation of what's going on.

Compiling the following code (reduced from `examples/draggable.links`) with `recheck_frontend_transformations=true`:
```
mutual {
  fun foo(x) client {
   receive { case Foo -> bar(x) }
  }

  fun bar(x) client {
   receive { case Foo -> foo(x) }
  }
}
```
produces a unification error:
```
Unification error: Rigid type variables 429 and 465 do not match
error: transformation pass 'Conditional(session_exceptions (typeable))' produced an ill-typed program.

../sandbox/links/T446.links:3: Type error: The function
    `bar'
has type
    `(429::Any) {:[|Foo|]|430}~> 431::Any'
while the arguments passed to it have types
    `465::Any'
and the currently allowed effects are
    `|hear:%478,wild|%486'
In expression: bar(x).
```
This further leads to IR typechecking error:
```
***: IR Type Error: Incompatible effects:
 Allowed:
  hear:[|Foo|],wild|380
 Actual:
  hear:[|Foo|],wild|%382
occurring in IR tail computation:
(Ir.Apply ((Ir.TApp ((Ir.Variable 231), [%381::Any; { |%382 }; %383::Any])),
   [(Ir.Variable 227)]))
```
Reproduced on 5d312665 with following config:
```
prelude=tests/empty_prelude.links
optimise=false
debug=true
recheck_frontend_transformations=true
hide_fresh_type_vars=false
show_flavours=true
show_quantifiers=true
print_types_pretty=true
show_raw_type_vars=true
```

**Comment by jstolarek** [comment]
Ok, I got this slightly one wrong. Enabling `recheck_frontend_transformations=true` reveals a true cause of the error in `session_exceptions` desugaring pass.

**Comment by jstolarek** [comment]
I reduced this further and this bug has nothing to do with session exceptions. This is a problem in type variable renamer, not session exceptions. (BTW frontend incorrectly reports offending pass name when the error is introduced by the typechecker. There's little we can do about this, but it's worth to be aware of.)

A minimal example to reproduce the bug is:

```
mutual {
  fun f(x) { g(x) }
  fun g(x) { f(x) }
}
```
Solution: don't rename type variables for mutually recursive functions. This does not seem to be necessary (see also @slindley's https://github.com/links-lang/links/issues/865#issuecomment-650766792) so I simply removed it. I would appreciate someone taking a look at the #884 PR.
